File: 02 백준/10 문자열/Python/Prob1764.py

"""
title  : 1764. 듣보잡 (Silver 4)
time   : 80ms        168ms
memory : 37264KB     115580KB
"""

import sys

def solution():

    input = sys.stdin.readline
    N, M = map(int, input().split())        # 듣도 못한 사람 N, 보도 못한 사람 M
    
    check = set()

    for _ in range(N):
        check.add(input().rstrip())
    
    answer = []
    for _ in range(M):
        name = input().rstrip()
        if name in check:
            answer.append(name)

    print(len(answer))
    for ans in sorted(answer):
        print(ans)

solution()

File: 02 백준/10 문자열/Java/src/Prob1764.java

/**
 * title  : 1764. 듣보잡 (Silver 4)
 * time   : 252ms
 * memory : 26136KB
 */

import java.io.*;
import java.util.*;

public class Prob1764 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        HashSet<String> nameSet = new HashSet<>();
        for (int i = 0; i < N; i++) {
            nameSet.add(br.readLine());
        }

        List<String> answer = new ArrayList<>();
        for (int i = 0; i < M; i++) {
            String name = br.readLine();
            if (nameSet.contains(name)) {
                answer.add(name);
            }
        }

        Collections.sort(answer);

        sb.append(answer.size()).append("\n");
        for (String ans : answer) {
            sb.append(ans).append("\n");
        }

        System.out.println(sb);
    }
}